test/test_unistd.rs

```diff
@@ -104,6 +104,57 @@ mod linux_android {
     }
 }
 
+#[test]
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+fn test_setgroups() {
+    if !Uid::current().is_root() {
+        // setgroups() requires root
+        return
+    }
+
+    // Save the existing groups
+    let old_groups = getgroups().unwrap();
+
+    // Set some new made up groups
+    let groups = [Gid::from_raw(123), Gid::from_raw(456)];
+    setgroups(&groups).unwrap();
+
+    let new_groups = getgroups().unwrap();
+    assert_eq!(new_groups, groups);
+
+    // Revert back to the old groups
+    setgroups(&old_groups).unwrap();
```

Comment:
Shouldn't this have one last assertion following this to check that they were properly reverted?

Reply:
The `getgroups()` and `setgroups()` calls at the start and end of these tests aren't really part of the tests, they're just a small attempt to restore the state of the current process to what it was before the test ran. 😕